odoo/addons/base/models/res_lang.py

```diff
@@ -241,44 +283,43 @@ def _data_get(self, monetary=False):
         return grouping, thousands_sep, decimal_point
 
     @api.model
-    @tools.ormcache()
-    def get_available(self):
-        """ Return the available languages as a list of (code, url_code, name,
-            active) sorted by name.
-        """
-        langs = self.with_context(active_test=False).search([])
-        return langs.get_sorted()
+    def get_available(self) -> list[tuple]:
+        """ Return the available languages as a tuple of LangInfo sorted by name. """
+        return [
+            (info.code, info.url_code, info.name, info.active, info.flag_image_url)
+            for info in self.get_all_info_by('id').values()
+        ]
 
     def get_sorted(self):
         return sorted([(lang.code, lang.url_code, lang.name, lang.active, lang.flag_image_url) for lang in self], key=itemgetter(2))
 
-    @tools.ormcache('self.id')
-    def _get_cached_values(self):
-        self.ensure_one()
-        return {
-            'id': self.id,
-            'code': self.code,
-            'url_code': self.url_code,
-            'name': self.name,
-        }
-
-    def _get_cached(self, field):
-        return self._get_cached_values()[field]
+    def _get_cached(self, field: str):
+        return self.get_all_info_by('id')[self.id].get(field)
 
     @api.model
-    @tools.ormcache('code')
     def _lang_code_to_urlcode(self, code):
-        for c, urlc, name, *_ in self.get_available():
-            if c == code:
-                return urlc
-        return self._lang_get(code).url_code
+        if info := self.get_all_info_by('code').get(code):
+            return info.url_code
+        return False
 
     @api.model
-    @tools.ormcache()
-    def get_installed(self):
-        """ Return the installed languages as a list of (code, name) sorted by name. """
-        langs = self.with_context(active_test=True).search([])
-        return sorted([(lang.code, lang.name) for lang in langs], key=itemgetter(1))
+    def get_installed(self) -> list[tuple[str]]:
+        """ Return the installed/active languages as a list of (code, name) sorted by name. """
+        return [(code, info.name) for code, info in self.get_installed_info_by('code').items()]
+
+    @tools.ormcache('field')
+    def get_installed_info_by(self, field: str) -> dict[str, LangInfo]:
+        return tools.frozendict({info.get(field): info for info in self.get_all_info_by('id').values() if info.active})
+
+    @tools.ormcache('field')
+    def get_all_info_by(self, field: str) -> dict[str, LangInfo]:
+        assert field in ('id',)
+        if field == 'id':
+            langs = self.with_context(active_test=False).search([], order='name')
```

Comment:
Shouldn't this `search` be sudoed?
I also recommend using `search_fetch()` to fetch the fields in the same query.